.editorconfig

```diff
@@ -12,19 +12,107 @@ csharp_indent_switch_labels = true
 csharp_indent_case_contents = true
 csharp_indent_labels = one_less_than_current
 
-# Globalization rules
-dotnet_code_quality.CA1305.excluded_symbol_names = T:System.Byte|T:System.SByte|T:System.Int16|T:System.UInt16|T:System.Int32|T:System.UInt32|T:System.Int64|T:System.UInt64|T:System.String|T:System.Text.StringBuilder|T:System.Convert
+# Style rules
+# Can't be in .globalconfig because dotnet format doesn't respect that https://github.com/dotnet/format/issues/1643
 
-dotnet_diagnostic.CA1305.severity = error
-dotnet_diagnostic.CA2101.severity = suggestion
+# Remove unnecessary cast
+dotnet_diagnostic.IDE0004.severity = warning
+# Remove unnecessary import
+dotnet_diagnostic.IDE0005.severity = warning
+# Use var instead of explicit type
+dotnet_diagnostic.IDE0007.severity = suggestion

```

Comment:
we should prefer one or the other of IDE0007 and IDE0008, IMO at error level to require consistency through the repo but at the least one of them should be preferred with none

Reply:
IDE0007 and IDE0008 are controlled through the variables `csharp_style_var_for_built_in_types`, `csharp_style_var_when_type_is_apparent` and `csharp_style_var_elsewhere`.
They are all set to `false` by default and `csharp_style_var_when_type_is_apparent` and `csharp_style_var_elsewhere` are configured to `true` below.

So the current severity levels will *suggest* using the explicit type when it's a builtin type and *suggest* using var when the type is apparent or not built-in.

We could probably bump these levels *eventually*, but that will require consensus from all active developers to make sure everyone agrees on these rules 100%, otherwise we'll just get edit wars. As is, great parts of the code are not consistent at all, which makes sense because these style rules have never been established.

Reply:
ok this makes more sense now, thanks